tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -141,13 +141,13 @@ static PyObject * THPVariable_arange(PyObject* self, PyObject* args, PyObject* k
       const auto options = TensorOptions()
           .dtype(scalarType)
           .device(r.device(6))
-          .layout(r.layout(5).layout)
+          .layout(r.layout(5))
           .requires_grad(r.toBool(8))
           .pinned_memory(r.toBool(7));
       return wrap(dispatch_arange(start, end, step, options));
     } else {
       TORCH_CHECK(!r.toBool(7), " `pin_memory` and `out` parameters are incompatible");
-      check_out_type_matches(r.tensor(3), r.scalartype(4), r.isNone(4), r.layout(5), r.isNone(5),
+      check_out_type_matches(r.tensor(3), r.scalartype(4), r.isNone(4), r.layout(5),
```

Comment:
@ezyang  Does here look alright? I didn't call `r.isNone` on layout. But `check_out_type_matched` takes an optional of Layout, and in the function it check if it's initialized.

Reply:
Yes, this is fine. The other calls should get updated to also call Optional too, but that's a change for another day.